main.rb

```diff
@@ -0,0 +1,123 @@
+      yield(arr[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result = []
+    my_each { |element| result.push element if yield element }
+    result
+  end
+
+  # rubocop:disable  Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |element| return false unless yield(element) }
+    elsif arg.class == Class
+      my_each { |element| return false unless element.class.ancestors.include? arg }
+    elsif arg.class == Regexp
+      my_each { |element| return false unless element =~ arg }
+    elsif arg.nil?
+      my_each { |element| return false unless element }
+    else
+      my_each { |element| return false unless element == arg }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |element| return true if yield element }
+    elsif arg.class == Class
+      my_each { |element| return true if element.class.ancestors.include? arg }
+    elsif arg.class == Regexp
+      my_each { |element| return true if arg =~ element }
+    elsif arg.nil?
+      my_each { |element| return true if element }
+    else
+      my_each { |element| return true if element == arg }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    if block_given?
+      my_each { |element| return false if yield element }
+    elsif arg.class == Class
+      my_each { |element| return false if element.class.ancestors.include? arg }
+    elsif arg.class == Regexp
+      my_each { |element| return false if arg =~ element }
+    elsif arg.nil?
+      my_each { |element| return false if element }
+    else
+      my_each { |element| return false if element == arg }
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    c = 0 # c - count
+    if block_given?
+      my_each { |element| c += 1 if yield(element) }
+    elsif arg.nil?
+      my_each { |_element| c += 1 }
+    else
+      my_each { |element| c += 1 if element == arg }
+    end
+    c
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    result = []
+    if self.class == Range
+      to_a.my_each_with_index { |element, index| result[index] = yield element }
+    else
+      my_each_with_index { |element, index| result[index] = yield element }
+    end
+    result
+  end
+
+  def my_inject(arg = nil, arg2 = nil)
+    output = is_a?(Range) ? min : self[0]
+    if block_given?
+      my_each_with_index { |ele, i| output = yield(output, ele) if i.positive? }
+      output = yield(output, arg) if arg
+    elsif arg.is_a?(Symbol) || arg.is_a?(String)
+      my_each_with_index { |ele, i| output = output.send(arg, ele) if i.positive? }
+    elsif arg2.is_a?(Symbol) || arg2.is_a?(String)
+      my_each_with_index { |ele, i| output = output.send(arg2, ele) if i.positive? }
+    elsif arg
+      return "my_inject: #{arg} is not a symbol nor a string"
+    else
+      return 'my_inject: no block given'
+    end
+    output
+  end
+end
```

Comment:
enumerables my_inject when a symbol is specified combines each element of the collection by applying the symbol as a named method
     Failure/Error: expect(range.my_inject(2, : *)).to eq range.inject(2, : *)

[**Note**] There should be no space between colon and the symbol. I added it for convenience, so that the symbol appears properly. 
     
       expected: 2534507766809448170301050680505397403698136797413376000000000000
            got: 1267253883404724085150525340252698701849068398706688000000000000

I will suggest you go through **```inject```** method in this [document](https://ruby-doc.org/core-2.6.5/Enumerable.html).

Reply:
Thank you soo much for this wonderful review. I will go through it again and fix it 🙏

Reply:
Code updated and it passes the following tests
```
puts (5..10).my_inject { |sum, n| sum + n } == (5..10).inject { |sum, n| sum + n } #=> 45
# Multiply some numbers
puts (5..10).my_inject(1, :*) == (5..10).inject(1, :*) #=> 151200
# Same using a block
puts (5..10).my_inject(1) { |product, n| product * n } == (5..10).inject(1) { |product, n| product * n } #=> 151200
# find the longest word
longest = %w[cat sheep bear].my_inject do |memo, word|
  memo.length > word.length ? memo : word
end
puts longest #=> "sheep"
puts (1..10).my_inject(2, :*) == (1..10).inject(2, :*)
```